### PR TITLE
[content-init] do not hide error during git checkout

### PR DESCRIPTION
**Pull request by sagor999**
## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/11854

## Release Notes

```release-note
none
```

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

File: components/content-service/pkg/initializer/git.go [hunk 1]
```diff
@@ -144,12 +144,14 @@ func (ws *GitInitializer) realizeCloneTarget(ctx context.Context) (err error) {
 	if ws.TargetMode == RemoteBranch {
 		// create local branch based on specific remote branch
 		if err := ws.Git(ctx, "checkout", "-B", ws.CloneTarget, "origin/"+ws.CloneTarget); err != nil {
-			log.WithField("remoteURI", ws.RemoteURI).WithField("branch", ws.CloneTarget).Debug("Remote branch doesn't exist.")
-			return nil
+			log.WithField("remoteURI", ws.RemoteURI).WithField("branch", ws.CloneTarget).Error("Remote branch doesn't exist.")
+			err = checkGitStatus(err)
```
Comment (jenting) [inline components/content-service/pkg/initializer/git.go:148]: Not understand why we check git statu at here 🤔 
https://github.com/gitpod-io/gitpod/blob/7640f4baf33c6840e3f5739f60214622b7db9b3c/components/content-service/pkg/initializer/git.go#L186-L194

Comment (sagor999) [reply components/content-service/pkg/initializer/git.go:148]: My understanding is that any git checkout command can encounter that Cloudflare error code, so I think we should handle it in every case?

File: components/content-service/pkg/initializer/git.go [hunk 2]
```diff
@@ -144,12 +144,14 @@ func (ws *GitInitializer) realizeCloneTarget(ctx context.Context) (err error) {
 	if ws.TargetMode == RemoteBranch {
 		// create local branch based on specific remote branch
 		if err := ws.Git(ctx, "checkout", "-B", ws.CloneTarget, "origin/"+ws.CloneTarget); err != nil {
-			log.WithField("remoteURI", ws.RemoteURI).WithField("branch", ws.CloneTarget).Debug("Remote branch doesn't exist.")
-			return nil
+			log.WithField("remoteURI", ws.RemoteURI).WithField("branch", ws.CloneTarget).Error("Remote branch doesn't exist.")
```
Comment (utam0k) [inline components/content-service/pkg/initializer/git.go:147]:
```suggestion
			log.WithError(err).WithField("remoteURI", ws.RemoteURI).WithField("branch", ws.CloneTarget).Error("Remote branch doesn't exist.")
```

File: components/content-service/pkg/initializer/git.go (hunk 1 above)
Comment (jenting) [reply components/content-service/pkg/initializer/git.go:148]: Sounds reasonable. 💯

**Review by jenting — APPROVED** [review]

**Pull request merged**